src/app/clusters/fan-control-server/fan-control-server.cpp

```diff
@@ -73,37 +115,73 @@ MatterFanControlClusterServerPreAttributeChangedCallback(const ConcreteAttribute
 
     switch (attributePath.mAttributeId)
     {
-    case SpeedSetting::Id:
-        // Check if the SpeedSetting is null.
-        if (NumericAttributeTraits<uint8_t>::IsNullValue(*value))
+    case FanMode::Id: {
+        if (*value == to_underlying(FanModeType::kOn))
         {
-
-            if (gWriteFromClusterLogic)
+            *value = static_cast<uint8_t>(FanModeType::kHigh);
```

Comment:
Please use `to_underlying`?

Also, it's somewhat unexpected that the pre-change callback lets you mutate the value.  I don't think anything guarantees that continuing to work....  Would it work to instead write FanModeType::kHigh and return WriteIgnored for the kOn write?